cpp/ql/test/query-tests/Likely Bugs/Format/WrongNumberOfFormatArguments/syntax_errors.c

```diff
@@ -3,5 +3,16 @@
 extern int printf(const char *fmt, ...);
 
 void test_syntax_error() {
-    printf("Error code %d: " FMT_MSG, 0, "");
+    // GOOD
+    printf("Error code %d: " UNDEFINED_MACRO, 0, "");
+
+    // GOOD
+    printf("%d%d",
+        (UNDEFINED_MACRO)1,
+        (UNDEFINED_MACRO)2);
+
+    // GOOD [FALSE POSITIVE]
+    printf("%d%d"
+        UNDEFINED_MACRO,
```

Comment:
I assume this is an FP, because it neither occurs on the line with the literal, not occurs in any of the arguments?

Reply:
Yes, so this is one of the FPs we saw on DCA that the current query or fixed query fails to catch. The problem is that the syntax error occurs on line 16, whereas the literal and function call are on line 15, and there is no obvious way to associate the two. 

There is no `ErrorExpr` in the function call, and the extent of the function call is only line 15. So right now, I can't think of a good way to detect this case.

It looks like EDG's syntax error recovery means dropping all of the expressions until the next `)`, which is pretty reasonable.